perfkitbenchmarker/vm_util.py

```diff
@@ -238,42 +240,55 @@ def RunThreaded(target, thread_params, max_concurrent_threads=200):
     return
 
   if not isinstance(thread_params[0], tuple):
-    thread_params = [((arg,), {}) for arg in thread_params]
+    target_arg_tuples = [(target, (arg,), {}) for arg in thread_params]
   elif (not isinstance(thread_params[0][0], tuple) or
         not isinstance(thread_params[0][1], dict)):
     raise ValueError('If Param is a tuple, the tuple must be (tuple, dict)')
   else:
-    thread_params = thread_params[:]
-
-  threads = []
-  exceptions = []
-  while thread_params and len(threads) < max_concurrent_threads:
-    args, kwargs = thread_params.pop()
-    thread = ThreadWithExceptions(target=target, args=args, kwargs=kwargs)
-    threads.append(thread)
+    target_arg_tuples = [(target, args, kwargs)
+                         for args, kwargs in thread_params]
+
+  queue = Queue.Queue()
+  log_context = log_util.GetThreadLogContext()
+  max_concurrent_threads = min(max_concurrent_threads, len(target_arg_tuples))
+  results = [None] * len(target_arg_tuples)
+  error_strings = []
+  for call_id in xrange(max_concurrent_threads):
+    target_arg_tuple = target_arg_tuples[call_id]
+    thread = threading.Thread(
```

Comment:
I think we've used default naming so far. RunThreaded can be called from within a function invoked by RunThreaded, so a name based on the call_id in this function wouldn't be good. An ideal name might be based on what the thread's role is, but the implementation for that would be complicated. I think it's best to delay on that and add it as a separate feature.